Kununu/ArchitectureSniffer/Configuration/Rules/MustBeFinal.php

```diff
@@ -4,37 +4,44 @@
 namespace Kununu\ArchitectureSniffer\Configuration\Rules;
 
 use InvalidArgumentException;
+use Kununu\ArchitectureSniffer\Configuration\ArchitectureLibrary;
 use Kununu\ArchitectureSniffer\Configuration\Selector\InterfaceClassSelector;
 use Kununu\ArchitectureSniffer\Configuration\Selector\Selectable;
 use PHPat\Selector\Selector;
 use PHPat\Test\Builder\Rule as PHPatRule;
 use PHPat\Test\PHPat;
 
-final readonly class MustBeFinal implements Rule
+final readonly class MustBeFinal extends AbstractRule
 {
-    public const string KEY = 'final';
+    public static function createRule(
+        string $groupName,
+        ArchitectureLibrary $library,
+    ): PHPatRule {
+        $includes = self::checkIfClassSelectors($library->getIncludesByGroup($groupName));
+        $excludes = $library->getExcludesByGroup($groupName);
 
-    public function __construct(public Selectable $selector)
-    {
-    }
+        $rule = PHPat::rule()->classes(...self::getPHPSelectors($includes));
 
-    public static function fromArray(Selectable $selector): self
-    {
-        if ($selector instanceof InterfaceClassSelector) {
-            throw new InvalidArgumentException(
-                'The class must not be an interface.'
-            );
-        }
+        $excludes = self::getPHPSelectors($excludes);
+        $excludes[] = Selector::isInterface();
+        $rule = $rule->excluding(...$excludes);
 
-        return new self($selector);
+        return $rule->shouldBeFinal()->because("$groupName must be final.");
     }
 
-    public function getPHPatRule(): PHPatRule
+    /**
+     * @param iterable<Selectable> $selectors
+     *
+     * @return iterable<Selectable>
+     */
+    private static function checkIfClassSelectors(iterable $selectors): iterable
     {
-        return PHPat::rule()
-            ->classes($this->selector->getPHPatSelector())
-            ->excluding(Selector::isInterface())
-            ->shouldBeFinal()
-            ->because("{$this->selector->getName()} must be final.");
+        foreach ($selectors as $selector) {
+            if ($selector instanceof InterfaceClassSelector) {
+                $name = $selector->interface;
+                throw new InvalidArgumentException("$name must be a class selector for rule MustBeFinal.");
+            }
+            yield $selector;
```

Comment:
Could also just return `$selectors` after the `foreach` instead of re-`yield`ing